stories/list-grid/list-grid.stories.js

```diff
@@ -0,0 +1,1880 @@
+                    </div>
+
+                    <div class="fd-grid-list__item--body fd-grid-list__item--body--padding">
+                        <div class="grid-list-item-body-example">
+                            <span class="fd-avatar fd-avatar--s grid-list-item-body-example-avatar" role="presentation" aria-label="John Doe"></span>
+                            <div>
+                                <h4 class="fd-title fd-title--h4">John Li</h4>
+                                <p>Product Owner, Company B</p>
+
+                                <div>
+                                    <p>781 Main Street</p>
+                                    <p>Anytown, SD 57401</p>
+                                    <p>USA</p>
+                                </div>
+
+                                <a href="#" class="fd-link" tabindex="0">john_li@example.com</a>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div  class="fd-col fd-col--12 fd-grid-list__group-header">
+                State: Error
+            </div>
+
+            <div  class="fd-col fd-col--12 fd-col-md--6 fd-col-lg--4 fd-col-xl--3">
+                <div tabindex="0" class="fd-grid-list__item fd-grid-list__item--state-error">
+                    <span class="fd-grid-list__highlight--error"></span>
+
+                    <div class="fd-toolbar fd-grid-list__toolbar">
+                        <span class="fd-toolbar__spacer"></span>
+
+                        <button class="fd-button fd-button--compact fd-button--transparent" aria-label="Edit">
+                            <i class="sap-icon--edit"></i>
+                        </button>
+                    </div>
+
+                    <div class="fd-grid-list__item--body fd-grid-list__item--body--padding">
+                        <div class="grid-list-item-body-example">
+                            <span class="fd-avatar fd-avatar--s grid-list-item-body-example-avatar" role="presentation" aria-label="John Doe"></span>
+                            <div>
+                                <h4 class="fd-title fd-title--h4">John Li</h4>
+                                <p>Product Owner, Company B</p>
+
+                                <div>
+                                    <p>781 Main Street</p>
+                                    <p>Anytown, SD 57401</p>
+                                    <p>USA</p>
+                                </div>
+                                
+
+                                <a href="#" class="fd-link" tabindex="0">john_li@example.com</a>
+                            </div>
+                        </div>
+
+                        <div class="fd-grid-list__item--state-actions">
+                            <span class="fd-object-status fd-object-status--negative">
+                                <span class="fd-object-status__text">(Contains errors)</span>
```

Comment:
The only problem with this layout is that the error state is not read out until the end of the content. So a screen reader user would not know unless they sat there to listen to the entire card content until it got to this line. Ideally the validation state should be announced first.

Reply:
fixed